kernelci/build.py

```diff
@@ -492,6 +492,512 @@ def _run_make(kdir, arch, target=None, jopt=None, silent=True, cc='gcc',
+
+    def _reset_bmeta(self):
+        if os.path.exists(self._bmeta_path):
+            os.unlink(self._bmeta_path)
+        if os.path.exists(self._steps_path):
+            os.unlink(self._steps_path)
+
+    def _load_bmeta(self):
+        if os.path.exists(self._bmeta_path):
+            with open(self._bmeta_path) as json_file:
+                self._bmeta = json.load(json_file)
+        if os.path.exists(self._steps_path):
+            with open(self._steps_path) as json_file:
+                self._steps = json.load(json_file)
+
+    def _add_run_step(self, name, jopt=None, status=None):
+        run_data = {
+            'name': name,
+            'start_time': self._start_time,
+            'duration': time.time() - self._start_time,
+            'cpus': self._get_cpus(),
+        }
+        if jopt is not None:
+            run_data['threads'] = str(jopt)
+        if status is not None:
+            run_data['status'] = "PASS" if status is True else "FAIL"
+        if self._log_path and os.path.exists(self._log_path):
+            run_data['log_file'] = os.path.basename(self._log_path)
+        self._steps.append(run_data)
+
+        total_duration = sum(s['duration'] for s in self._steps)
+        all_status = set(s['status'] for s in self._steps)
+        self._bmeta['build'] = {
+            'duration': total_duration,
+            'status': 'PASS' if all_status == {'PASS'} else 'FAIL'
+        }
+
+    def _save_bmeta(self):
+        with open(self._bmeta_path, 'w') as json_file:
+            json.dump(self._bmeta, json_file, indent=4, sort_keys=True)
+        with open(self._steps_path, 'w') as json_file:
+            json.dump(self._steps, json_file, indent=4, sort_keys=True)
+
+    def _get_cpus(self):
+        cpus = {}
+        if os.path.exists('/proc/cpuinfo'):
+            cpu_list = []
+            with open('/proc/cpuinfo') as f:
+                for line in f:
+                    if 'model name' in line:
+                        cpu_list.append(line.split(':')[1].strip())
+            for cpu in cpu_list:
+                cpu_entry = cpus.setdefault(cpu, 0)
+                cpus[cpu] += 1
+        return cpus
+
+    def _kernel_config_enabled(self, config_name):
+        dot_config = os.path.join(self._output_path, '.config')
+        cmd = 'grep -cq CONFIG_{}=y {}'.format(config_name, dot_config)
+        return shell_cmd(cmd, True)
+
+    def _output_to_file(self, cmd, log_file, rel_path=None):
+        open(log_file, 'a').write("#\n# {}\n#\n".format(cmd))
+        if rel_path:
+            log_file = os.path.relpath(log_file, rel_path)
+        cmd = "/bin/bash -c '(set -o pipefail; {} 2>&1 | tee -a {})'".format(
+            cmd, log_file)
+        return cmd
+
+    def _get_make_opts(self, opts, make_path):
+        env = self._bmeta['environment']
+        make_opts = env['make_opts'].copy()
+        if opts:
+            make_opts.update(opts)
+
+        arch = env['arch']
+        make_opts['ARCH'] = arch
```

Comment:
```suggestion
        make_opts['ARCH'] = env['arch']
```

Reply:
The `arch` variable is used further down for `ccache`.